daffodil-io/src/main/scala/org/apache/daffodil/io/processors/charset/BitsCharsetDefinition.scala

```diff
@@ -16,13 +16,18 @@
  */
 package org.apache.daffodil.io.processors.charset
 
+import org.apache.daffodil.lib.util.hasNameDefined
+
 /**
  * These are the classes which must be dynamically loaded in order to add a charset implementation
  * to Daffodil. All charsets must implement this class and be added to the 
  * org.apache.daffodil.runtime1.processors.charset.BitsCharsetDefinition file in 
  * daffodil-io/src/main/resources/META-INF/services. name() must return a fully capitalized string
  */
-abstract class BitsCharsetDefinition(charset: BitsCharset, alias: Option[String] = None) {
+abstract class BitsCharsetDefinition(
+  charset: BitsCharset,
+  alias: Option[String] = None
+) extends hasNameDefined {
```

Comment:
Should be "H" initial caps on  trait name. 

I suggest remove the suffix "Defined" and just call it "HasName" or "HasNameMethod"